### PR TITLE
refactor(modal): restructure keyboard support

### PR DESCRIPTION
**Pull request by eweseong**
https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

- restructure keyboard support to rely on react keyboard event over native keyboard event
- replace explicit focus trap element with programmatical approach

### CODE REVIEW

File: lib/src/Modal/Modal.tsx [hunk 1]
```diff
@@ -23,21 +24,45 @@ export type ModalProps = JSX.IntrinsicElements["div"] & {
     position?: ModalPosition;
     /** Modal toggle */
     toggle?: boolean;
-    /** To only allow tab to focus within modal */
-    trapFocus?: boolean;
```
Comment (kherP) [inline lib/src/Modal/Modal.tsx:27]: before removing it entirely, should we just mark it @deprecated and keep it until next major release?

File: docs/src/pages/docs/modal.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ const positions: Array<DynamicFormOption<ModalPosition>> = [
 
 const NotificationPage: React.FC = () => {
     const [toggle, setToggle] = React.useState<boolean>(false);
+    const toggleButtonRef = React.useRef<HTMLButtonElement>();
```
Comment (mario-subo) [inline docs/src/pages/docs/modal.tsx:29]: missing type

File: lib/src/Modal/Modal.tsx (hunk 1 above)
Comment (eweseong) [reply lib/src/Modal/Modal.tsx:27]: if I've added in a breaking change commit 57d231c1421c00e4b4946f2322d7bd7530a92e76 while working on one of the accessibility changes, would this bump our major version while preping the next release?

Comment (eweseong) [reply lib/src/Modal/Modal.tsx]: after discussion with the team, the `trapFocus` prop would be deprecated regardless if the major version were bumped

**Review by kherP — APPROVED** [review]

**Review by mario-subo — APPROVED** [review]

**Pull request merged**